src/views/virtualmachines/actions/components/VirtualMachineMigration/hooks/useMigrationState.ts

```diff
@@ -0,0 +1,56 @@
+import { useState } from 'react';
+
+import { IoK8sApiCoreV1PersistentVolumeClaim } from '@kubevirt-ui/kubevirt-api/kubernetes';
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+
+import { MigMigration } from '../constants';
+
+import { migrateBulkVMs } from './migrateBulkVMs';
+import { migrateVM } from './migrateVM';
+
+type UseMigrationState = (
+  vms: V1VirtualMachine[],
+  namespacePVCs: IoK8sApiCoreV1PersistentVolumeClaim[],
+  pvcsToMigrate: IoK8sApiCoreV1PersistentVolumeClaim[],
+  destinationStorageClass: string,
+) => {
+  migMigration?: MigMigration;
+  migrationError: Error;
+  migrationLoading: boolean;
+  migrationStarted: boolean;
+  onSubmit: () => Promise<void>;
+};
+
+const useMigrationState: UseMigrationState = (
+  vms,
+  namespacePVCs,
+  pvcsToMigrate,
+  destinationStorageClass,
+) => {
+  const [migrationError, setMigrationError] = useState<Error | null>(null);
+  const [migrationLoading, setMigrationLoading] = useState(false);
+  const [migrationStarted, setMigrationStarted] = useState(false);
+  const [migMigration, setMigMigration] = useState<MigMigration>(null);
+
+  const onSubmit = async () => {
+    setMigrationLoading(true);
+    setMigrationError(null);
+    try {
+      vms.length === 1
+        ? await migrateVM(vms?.[0], pvcsToMigrate, destinationStorageClass)
+        : setMigMigration(
```

Comment:
shouldnt this be setMigrations ?

Reply:
why not waiting for result then set in state? takes too much time?
i don't think the catch block has any effect on the migrateBulkVms as u set it in a state

Reply:
I do `setMigMigration` as the resource is `MigMigration` but if you want i can change that.